drink/src/runtime/pallet_contracts_debugging.rs

```diff
@@ -0,0 +1,114 @@
+//! This module provides all the necessary elements for supporting contract debugging directly in
+//! the contracts pallet.
+//!
+//! # Smart-contract developer <-> pallet-contracts interaction flow
+//!
+//! The interaction between end-user and runtime is as follows:
+//! 1. At some points during execution, the pallet invokes some callback through its configuration
+//! parameter `Debug`.
+//! 2. In order to forward the callback outside the runtime, `Debug` will call a runtime interface,
+//! that will then forward the call further to the proper runtime extension.
+//! 3. The runtime extension can be fully controlled by the end-user. It just has to be registered
+//! in the runtime.
+//!
+//! So, in brief: pallet-contracts -> runtime interface -> runtime extension
+//!              |<-----------runtime side-------------->|<---user side--->|
```

Comment:
Is runtime extension really on the "user" side? I guess it depends who the user here is - I'd say it's _drink!_ and not drink's user? Or more generally - user is where the set up of the runtime happens. Here that's drink!'s `Sandbox` and in normal chain that would be a node.

Is the above correct?

Reply:
not necessary - any drink! user can provide their own extension that would express their own intention (and possible could even capture some local environment)